source/configuration/modules/omfile.rst

```diff
@@ -1,434 +1,437 @@
-   set to "off", the error will be ignored and processing continues.
-   Keep in mind, that the files in this case may be (in)accessible by
-   people who should not have permission. The default is "on".
-
-.. function::  createDirs [switch]
-
-   *Default: on*
-
-   create directories on an as-needed basis
-
-.. function::  sync [switch]
-
-   *Default: off*
-
-   enables file syncing capability of omfile.
-
-   When enabled, rsyslog does a sync to the data file as well as the
-   directory it resides after processing each batch. There currently
-   is no way to sync only after each n-th batch.
-
-   Enabling sync causes a severe performance hit. Actually,
-   it slows omfile so much down, that the probability of loosing messages
-   **increases**. In short,
-   you should enable syncing only if you know exactly what you do, and
-   fully understand how the rest of the engine works, and have tuned
-   the rest of the engine to lossless operations.
-
-.. function::  sig.provider [providerName]
-
-   *Default: no signature provider*
-
-   Selects a signature provider for log signing. By selecting a provider,
-   the signature feature is turned on.
-
-   Currently, there are two providers available ":doc:`gt <sigprov_gt>`" and ":doc:`ksi <sigprov_ksi>`". 
-
-.. function::  cry.provider [providerName]
-
-   *Default: no crypto provider*
-
-   Selects a crypto provider for log encryption. By selecting a provider,
-   the encryption feature is turned on.
-
-   Currently, there only is one provider called ":doc:`gcry <../cryprov_gcry>`".
-
-See Also
---------
-
-- `Sign log messages through signature provider
-  Guardtime <http://www.rsyslog.com/how-to-sign-log-messages-through-signature-provider-guardtime/>`_
-
-Caveats/Known Bugs
-------------------
-
--  people often report problems that dynafiles are not properly created.
-   The common cause for this problem is SELinux rules, which do not permit
-   the create of those files (check generated file names and pathes!). The
-   same happens for generic permission issues (this is often a problem
-   under Ubuntu where permissions are dropped by default)
-
--  One needs to be careful with log rotation if signatures and/or
+if turned *on*, the files will be written in asynchronous mode via a
+separate thread. In that case, double buffers will be used so that
+one buffer can be filled while the other buffer is being written.
+Mind that if using *dynaFile* and this parameter is turned *on*, it can result in **a lot** of rsyslog threads (one per file currently open).
+   
+Note that in order to enable *flushInterval*, this parameter must be *on*. Otherwise, the flush interval will be ignored.
+Also note that when flushOnTXEnd is *on* but asyncWriting is *off*, output will only be written when the buffer is full.
+This may take several hours, or even require a rsyslog shutdown. However, a buffer flush can be forced in that case by sending rsyslogd a HUP signal.
+
```

Comment:
possibly add a note that the most common reason to use this is to improve performance of zipped files